src/sage/algebras/weyl_algebra.py

```diff
@@ -1123,3 +1141,519 @@ def _act_(self, g, x):
+
+            sage: W.<x> = DifferentialWeylAlgebra(QQ, n=oo)
+            sage: W.inject_variables()
+            Defining x, dx
+            sage: W.differential(1)
+            dx[1]
+            sage: W.differential(1) == dx[1]
+            True
+        """
+        return self.element_class(self, {(self._var_index.one(),
+                                          self._diff_index.gen(i)): self.base_ring().one()})
+
+    @cached_method
+    def differentials(self):
+        r"""
+        Return the differential generators of ``self``.
+
+        OUTPUT: A (lazy) family containing the differential generators `\partial_{x_i}`
+
+        EXAMPLES::
+
+            sage: W.<x> = DifferentialWeylAlgebra(QQ, n=oo)
+            sage: dx = W.differentials(); dx
+            Lazy family (dx(i))_{i in Non negative integers}
+            sage: dx[3] == W.differential(3)
+            True
+        """
+        return Family(NonNegativeIntegers(), lambda x: self.differential(x), name=self.variable_names()[1])
+
+    @cached_method
+    def zero(self):
+        """
+        Return the additive identity element `0` of ``self``.
+
+        EXAMPLES::
+
+            sage: W.<x> = DifferentialWeylAlgebra(QQ, n=oo)
+            sage: x[1] + W.zero() == x[1]
+            True
+            sage: x[1]*W.zero() == W.zero()
+            True
+        """
+        return self.element_class(self, {})
+
+    @cached_method
+    def one_basis(self):
+        """
+        Return the multiplicative identity element `1` of ``self``.
+
+        EXAMPLES::
+
+            sage: W.<x> = DifferentialWeylAlgebra(QQ, n=oo)
+            sage: W.one_basis() == W.one().leading_support()
+            True
+            sage: x[1]*W.one() == x[1]
+            True
+        """
+        return (self._var_index.one(), self._diff_index.one())
+
+    @cached_method
+    def basis(self):
+        """
+        Return a basis for ``self``.
+
+        EXAMPLES::
+
+            sage: W.<x> = DifferentialWeylAlgebra(QQ, n=oo)
+            sage: B = W.basis(); B
+            Lazy family (basis map(i))_{i in The Cartesian product of
+             (Free abelian monoid indexed by Non negative integers,
+             Free abelian monoid indexed by Non negative integers)}
+            sage: idx = W._var_index
+            sage: B[(idx.an_element(), idx.one())]
+            x[0]*x[1]^2*x[2]^3*x[42]
+        """
+        index_set = cartesian_product([self._var_index, self._diff_index])
+        one = self.base_ring().one()
+        return Family(index_set,
+                      lambda x: self.element_class(self, {(x[0], x[1]): one}),
+                      name='basis map')
+
+    def degree_on_basis(self, x):
+        """
+        Return the degree of basis element indexed by ``x``. This is the total
+         degree of the polynomial and differential parts of ``x``.
+
+        INPUT:
+
+        - ``x`` -- an index for a basis element
+
+        OUTPUT: A nonnegative integer
+
+        EXAMPLE::
```

Comment:
```suggestion
        EXAMPLES::
```
We generally make it plural, even if there is only 1 test/example.